proto/common/common.proto

```diff
@@ -67,11 +67,16 @@ message Eos {
 message TapEvent {
   TcpAddress source = 1;
   TcpAddress target = 2;
+  EndpointMeta target_meta = 4;
```

Comment:
Does `TapEvent.target_meta` describe the source or destination (or both) of the tap event? If the destination, how does it relate to `K8sTapRequest.match.destinations` and `ObserveRequest.destination_label`? Are they all destination labels? If so, consider naming them similarly?

Related, should we change `ResourceSelection.label_selector` to be of type `map<string, string>`?

...or, going a step further, we could re-use k8s' `LabelSelector`:
```proto

// A label selector is a label query over a set of resources. The result of matchLabels and
// matchExpressions are ANDed. An empty label selector matches all objects. A null
// label selector matches no objects.
message LabelSelector {
  // matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
  // map is equivalent to an element of matchExpressions, whose key field is "key", the
  // operator is "In", and the values array contains only "value". The requirements are ANDed.
  // +optional
  map<string, string> matchLabels = 1;

  // matchExpressions is a list of label selector requirements. The requirements are ANDed.
  // +optional
  repeated LabelSelectorRequirement matchExpressions = 2;
}

// A label selector requirement is a selector that contains values, a key, and an operator that
// relates the key and values.
message LabelSelectorRequirement {
  // key is the label key that the selector applies to.
  // +patchMergeKey=key
  // +patchStrategy=merge
  optional string key = 1;

  // operator represents a key's relationship to a set of values.
  // Valid operators are In, NotIn, Exists and DoesNotExist.
  optional string operator = 2;

  // values is an array of string values. If the operator is In or NotIn,
  // the values array must be non-empty. If the operator is Exists or DoesNotExist,
  // the values array must be empty. This array is replaced during a strategic
  // merge patch.
  // +optional
  repeated string values = 3;
}
```

Reply:
> Related, should we change ResourceSelection.label_selector to be of type map<string, string>?

No, this is meant to represent a kubernetes label selector, which is not just a set of key-vals.

Reply:
> Does TapEvent.target_meta describe the source or destination (or both) of the tap event?

AIUI it is meant to mean the destination. We should update the terminology.

I'm skeptical that we want to take a dependency on k8s' proto, but yes, ResourceSelection should probably be replaced by this.

I don't think we should try to make EndpointMeta any more generic than it is... it's capturing details about a destination known to the proxy. right now that includes destination labels (not a selection).  In the future it may include other things.

I don't think we should change TapEvent, however. it's just reflecting metadata about the destination endpoint.

Reply:
I'm not sure if it's worth parsing out label selectors in the client, though. it seems like we can just treat that as a string to be validated on the server, bypassing the need for these added types